### ISSUE TITLE
Use pins instead of "mark as unread"

### ISSUE THREAD

**Issue by 12people** [k9mail/k-9#2621]
Right now, when I read a message and need to return to it, I mark it as unread in K9Mail. However, this has the unfortunate effect of increasing my unread count, so that I don't know how many messages are actually unread and how many are just waiting for me to return to them.

Google Inbox solves this quite nicely: instead of "Mark as unread", it has a pin feature, which lets one pin messages to return to. It also allows showing only pinned messages and archiving all messages except for pinned messages (for "inbox zero").

Please consider replacing "Mark as unread" with pinning.

**Comment by philipwhiuk** [comment]
I don't think removing the ability to mark unread is a good idea. Note Google does also allow you to mark unread. However adding pinning is something we could do.

One question is how to implement pinning - whether it's purely a client thing or whether it should be synched to the server as a flag.

If the later we should consider trying to use the same flag as other clients (e.g. Google Inbox) if possible.

**Comment by 12people** [comment]
@philipwhiuk Google doesn't allowing marking as unread in Inbox. (It does in Gmail, but there are no pins in Gmail.) Pins are a direct replacement for Mark as unread, it doesn't make sense to have both at once.

**Comment by 12people** [comment]
@AvoidMastodonBlue 
It's a different way of managing an inbox. Let me explain the difference between GMail's and Inbox's approach:

GMail:
*Unread state:* Unread messages and messages to look at soon (marked as unread)
*Starred messages:* Important messages that require quick access (bookmarks, basically)

Inbox:
*Unread state:* Unread messages only
*Pinned messages:* Messages to look at soon

Additionally, Inbox's UI centers around pinned messages. The headline feature is a prominent "sweep" button, which archives all messages that are not pinned (including unread ones). That makes it much easier to aim for inbox zero.

There's no such functionality with stars and, given their bookmark-like nature, wouldn't make as much sense as it does with pins.

**Comment by mikqu** [comment]
I would like to add my voice to this old request. I use stars, but they don't work well for emails I want pinned for a long time. I have several accounts in my unified inbox (default view). Starred emails stay on top only until the unified inbox fills up. Then I have to scroll down and select 'show 25 more' over and over until I restore my starred messages. Sometimes these are for travel bookings months in advance, or bills to pay. It would be really nice to pin them to the top for as long as I want them there :)

**Comment by ckaotik** [comment]
> It would be really nice to pin them to the top for as long as I want them there :)

I agree, maybe a setting to "Download and keep all pinned messages" would be possible? Already missed bills because of not being on my PC regularly and my phone swallowing pinned mails from earlier 🙄 Setting the app to synchronize all mails is not feasible for me.
